src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabMergeRequestSCMEvent.java

```diff
@@ -109,42 +114,42 @@ public String description() {
                 .newRequest(source, null)) {
             MergeRequestEvent.ObjectAttributes m = getPayload().getObjectAttributes();
             String originOwner = getPayload().getUser().getUsername();
+            // This is a hack to get the namespace from the path with namespace
             String originProjectPath = m.getSource().getPathWithNamespace();
-            Set<ChangeRequestCheckoutStrategy> strategies = request.getMRStrategies(
-                    source.getProjectPath().equalsIgnoreCase(originProjectPath)
-            );
-            for (ChangeRequestCheckoutStrategy strategy : strategies) {
-                MergeRequestSCMHead h = new MergeRequestSCMHead(
-                        "MR-" + m.getIid() + (strategies.size() > 1 ? "-" + strategy.name()
-                                .toLowerCase(Locale.ENGLISH) : ""),
-                        m.getId(),
-                        new BranchSCMHead(m.getTargetBranch()),
-                        ChangeRequestCheckoutStrategy.MERGE,
-                        originProjectPath.equalsIgnoreCase(source.getProjectPath())
-                                ? SCMHeadOrigin.DEFAULT
-                                : new SCMHeadOrigin.Fork(originProjectPath),
-                        originOwner,
-                        originProjectPath,
-                        m.getSourceBranch());
-                GitLabApi gitLabApi = GitLabHelper.apiBuilder(source.getServerName());
-                // Since rn there is no way to get the commit hash of target branch fetch the mr again
-                // A better way can be searching for the target branch head and getting the commit hash
-                MergeRequest mr = gitLabApi.getMergeRequestApi().getMergeRequest(source.getProjectPath(), m.getIid());
+            int namespaceLength = originProjectPath.lastIndexOf("/");
+            originProjectPath = originProjectPath.substring(0, namespaceLength);
```

Comment:
Seems to be that `originProjectPath` after this would be considered origin owner?
where `String originOwner = getPayload().getUser().getUsername();` is actually the user who one who just intiated the the webhook event?

Reply:
Yes, it seems. Removing the projectName from `originProjectPath` is wrong. It should be the whole path only. :man_facepalming:

Reply:
In the `originOwner` should be the `namespace path` (not including projectName).

Reply:
Fixed in d2eb34f